pkg/state/appender.go

```diff
@@ -630,6 +629,126 @@ func (a *txAppender) createInitialBlockSnapshot(minerAndRewardDiff txDiff) (txSn
 	return snapshot, nil
 }
 
+func (a *txAppender) applySnapshotsInLightNode(
+	params *appendBlockParams,
+	stateHash crypto.Digest,
+	hasher *txSnapshotHasher,
+) error {
+	blockInfo, errBlockInfo := a.currentBlockInfo()
+	if errBlockInfo != nil {
+		return errBlockInfo
+	}
+	for i, txs := range params.snapshot.TxSnapshots {
+		txID, idErr := params.transactions[i].GetID(a.settings.AddressSchemeCharacter)
+		if idErr != nil {
+			return idErr
+		}
+		if len(txs) == 0 { // sanity check
+			return errors.Errorf("snapshot of txID %q cannot be empty", base58.Encode(txID))
+		}
+
+		for _, snapshot := range txs {
+			if errApply := snapshot.Apply(a.txHandler.sa); errApply != nil {
+				return errors.Wrap(errApply, "failed to apply tx snapshot")
+			}
+		}
```

Comment:
`txSnapshot.Apply` method must be used for snapshot applying.
(It's not obvious, agree. Will think how can it be improved)